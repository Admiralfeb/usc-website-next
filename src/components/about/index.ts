export * from './allies';
export * from './faction';
export * from './hc';
export * from './rules';
export * from './carriers';
