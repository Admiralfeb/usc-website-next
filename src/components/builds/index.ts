export * from './builds/buildAdd';
export * from './builds/buildDetail';
export * from './queryandBuildList';
