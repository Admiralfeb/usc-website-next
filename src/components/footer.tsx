import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';

function ImgCopyright() {
  return (
    <Typography
      variant='body2'
      color='textSecondary'
      style={{ textAlign: 'center' }}>
      Thank you to CMDR RaZ uryel for the background image
    </Typography>
  );
}
function Copyright() {
  return (
    <Typography
      variant='body2'
      color='textSecondary'
      style={{ textAlign: 'center' }}>
      {'Copyright Â© '}
      <Link color='inherit' href='https://unitedsystems.org/'>
        United Systems Cooperative
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  footer: {
    padding: theme.spacing(3, 2),
    marginTop: 'auto',
    backgroundColor:
      theme.palette.type === 'light'
        ? theme.palette.grey[200]
        : theme.palette.grey[800],
  },
  center: {
    textAlign: 'center',
  },
}));

export const Footer = () => {
  const classes = useStyles();

  return (
    <footer className={classes.footer}>
      <Container maxWidth='sm' className={classes.center}>
        <Copyright />
        <ImgCopyright />
      </Container>
    </footer>
  );
};
