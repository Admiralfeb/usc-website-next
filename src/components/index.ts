export * from './home';
export * from './footer';
export * from './notFound';
export * from './auth/login';
