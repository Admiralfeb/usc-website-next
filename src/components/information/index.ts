export * from './infoSection';
export * from './infographic';
export * from './information';
export * from './miningMap';
export * from './shipReviews';
