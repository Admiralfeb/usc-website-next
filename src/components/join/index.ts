export * from './join';
export * from './joinList';
export * from './joinNew';
export * from './joinFormMember';
