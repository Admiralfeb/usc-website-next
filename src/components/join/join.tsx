import {
  Fade,
  Link,
  makeStyles,
  Paper,
  Typography,
  useMediaQuery,
  useTheme,
} from '@material-ui/core';
import { NavLink } from 'react-router-dom';
import { useLinks } from 'hooks/useLinks';

const useStyles = makeStyles({
  root: {
    height: '100%',
  },
  header: {
    textAlign: 'center',
  },
  paper: {
    textAlign: 'center',
    width: '80%',
    margin: 'auto',
    padding: 5,
    paddingBottom: 10,
    marginBottom: 5,
  },
  iframe: {
    width: '100%',
    height: '59vw',
  },
  step: {
    display: 'grid',
    gridTemplate: '1fr / 1fr 1fr',
  },
  right: {
    textAlign: 'right',
  },
  left: {
    textAlign: 'left',
  },
});

export const Join = () => {
  const classes = useStyles();
  const theme = useTheme();
  const isMedium = useMediaQuery(theme.breakpoints.up('md'));
  const { inaraSquadLink, discordLink } = useLinks();

  return (
    <Fade in={true}>
      <div className={classes.root}>
        <Typography variant="h3" className={classes.header}>
          Join Us!
        </Typography>
        <Paper className={classes.paper}>
          <Typography>
            Step 1: Read the rules on the{' '}
            <Link to="/about/rules" target="_blank" component={NavLink}>
              About Page
            </Link>
            .
          </Typography>
          {isMedium ? (
            <Typography>Step 2: Fill out the form below.</Typography>
          ) : (
            <Typography>
              Step 2: Fill out the{' '}
              <Link
                href="https://docs.google.com/forms/d/e/1FAIpQLSfw7vIL8FY8ZGTF8YFjn3x-zGdhP5k9J6JALXAjXA7_Mm3-Rw/viewform?usp=sf_link"
                target="_blank"
              >
                USC Join Form
              </Link>
              .
            </Typography>
          )}

          <Typography>
            Step 3: Join our{' '}
            <Link href={discordLink} target="_blank">
              Discord
            </Link>
            .
          </Typography>

          <Typography>
            Recommended: Find us in-game! (UCPC for PC | UCXB for Xbox | UCPS
            for PlayStation)
          </Typography>
          <Typography>
            Recommended: Join our{' '}
            <Link href={inaraSquadLink} target="_blank">
              Inara Squadron
            </Link>
            .
          </Typography>
          {isMedium && (
            <iframe
              className={classes.iframe}
              title="USC Application Form"
              id="joinFormContainer"
              src="https://docs.google.com/forms/d/e/1FAIpQLSfw7vIL8FY8ZGTF8YFjn3x-zGdhP5k9J6JALXAjXA7_Mm3-Rw/viewform?embedded=true"
            >
              Loadingâ€¦
            </iframe>
          )}
          {/* <iframe
          className={classes.iframe}
          title='Discord Widget'
          id='embeddedDiscord'
          src='https://discord.com/widget?id=662439414152167434&amp;theme=dark'
          allowTransparency
          frameBorder='0'
        sandbox='allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts'></iframe> */}
        </Paper>
      </div>
    </Fade>
  );
};
