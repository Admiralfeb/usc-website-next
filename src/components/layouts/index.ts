export * from './about';
export * from './primary';
