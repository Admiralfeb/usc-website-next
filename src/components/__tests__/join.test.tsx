import { cleanup, render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Join } from 'components/join';

let componentBody: RenderResult;

describe('Join', () => {
  beforeEach(() => {
    componentBody = render(
      <MemoryRouter initialEntries={['/join']}>
        <Join />
      </MemoryRouter>
    );
  });

  afterEach(cleanup);

  it(`should match the snapshot`, () => {
    const { baseElement } = componentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
