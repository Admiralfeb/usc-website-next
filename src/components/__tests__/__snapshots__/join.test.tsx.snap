// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-root-1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <h3
        class="MuiTypography-root makeStyles-header-2 MuiTypography-h3"
      >
        Join Us!
      </h3>
      <div
        class="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Step 1: Read the rules on the
           
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/about/rules"
            target="_blank"
          >
            About Page
          </a>
          .
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Step 2: Fill out the
           
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://docs.google.com/forms/d/e/1FAIpQLSfw7vIL8FY8ZGTF8YFjn3x-zGdhP5k9J6JALXAjXA7_Mm3-Rw/viewform?usp=sf_link"
            target="_blank"
          >
            USC Join Form
          </a>
          .
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Step 3: Join our
           
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://discord.gg/br8Fy2M6cv"
            target="_blank"
          >
            Discord
          </a>
          .
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Recommended: Find us in-game! (UCPC for PC | UCXB for Xbox | UCPS for PlayStation)
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Recommended: Join our
           
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://inara.cz/squadron/7028/"
            target="_blank"
          >
            Inara Squadron
          </a>
          .
        </p>
      </div>
    </div>
  </div>
</body>
`;
